### PR TITLE
improve hashing performance for uuid

### PR DESCRIPTION
**Pull request by KristofferC**
I was working with some dicts that has `UUID`s for keys and they felt kind of slow. This helps a bit:

Example benchmark:

```jl
using UUIDs, BenchmarkTools

function bench(uuids, numbers)
    d = Dict{UUID, Int}()
    sizehint!(d, length(uuids))
    for (uuid, number) in zip(uuids, numbers)
        d[uuid] = number
    end
    return d
end

N = 10^4
uuids = [uuid4() for i in 1:N]
numbers = rand(Int, N)
@btime bench(uuids, numbers)
```

Master:

```
 264.006 μs (7 allocations: 400.97 KiB)
```

PR:

```
 190.882 μs (7 allocations: 400.97 KiB)
```

### CODE REVIEW

**Review by Sacha0 — APPROVED** [review]

**Review by Sacha0 — APPROVED** [review]
Even better! :)

File: base/uuid.jl [hunk 1]
```diff
@@ -31,6 +31,11 @@ end
 
 UInt128(u::UUID) = u.value
 
+let
+    uuid_hash_seed = UInt === UInt64 ? 0xd06fa04f86f11b53 : 0x96a1f36d
+    hash(uuid::UUID, h::UInt) = hash(uuid_hash_seed, hash(convert(NTuple{2, UInt64}, uuid), h))
```
Comment (simeonschaub) [inline base/uuid.jl:36]: Doesn't this just create a new local function? :)

Comment (Sacha0) [reply base/uuid.jl:36]: Oh gosh, good catch 😄.

Comment (Sacha0) [reply base/uuid.jl:36]: I.e., it looks like this change follows the pattern of the following `let` blocks, in which case `hash` is missing a qualification that would lift this `hash` method from the `let`-block's scope? :)

Comment (simeonschaub) [reply base/uuid.jl:36]: I am not sure you can overload a function from inside a `let` block. The easiest solution is probably just to make `uuid_hash_seed` a global constant.

Comment (Sacha0) [reply base/uuid.jl:36]: The code just a bit below uses that pattern. See also e.g.
```
julia> struct FluffType end

julia> let
       Base.hash(::FluffType) = "fluffy"
       end

julia> hash(FluffType())
"fluffy"
```

Comment (KristofferC) [reply base/uuid.jl]: Oops, thanks.

**Pull request merged**